### ISSUE TITLE
argon2: `Argon2::hash_password` ignores provided configuration

### ISSUE THREAD

**Issue by philipahlberg** [RustCrypto/password-hashes#179]
When using `Argon::new`, and subsequently `<Argon as PasswordHasher>::hash_password`, the parameters provided in `Argon::new` are ignored, and default values are used instead:

```rust
use argon2::{Argon2, Version, PasswordHasher};

const PASSWORD: &'static str = "password";
const SECRET_KEY: &'static str = "secret_key";
const SALT: &'static str = "longenoughsalt";

let hasher = Argon2::new(
    Some(SECRET_KEY.as_bytes()),
    192,
    8196,
    4,
    Version::V0x10,
).unwrap(); 

hasher
    .hash_password_simple(PASSWORD.as_bytes(), SALT)
    .unwrap()
    .to_string()

// $argon2id$v=19$m=4096,t=3,p=1$longenoughsalt$sqRGsTa4Wz1kLgdImtaf3WfwDAYPp7VZf8wSHfn/KLk
//           ^^^^ ^^^^^^ ^^^ ^^^
```

**Comment by tarcieri** [comment]
Aha, good catch.

The problem is that the default implementation of `PasswordHasher::hash_password_simple` always uses the default parameters, then calls `PasswordHasher::hash_password`:

https://docs.rs/password-hash/0.2.1/src/password_hash/lib.rs.html#92-102

In the case of `argon2` crate, it seems that relationship should be reversed, with `PasswordHasher::hash_password` (which takes explicit params) calling `PasswordHasher::hash_password_simple`, and the latter using the configured params rather than the defaults.

**Comment by tarcieri** [comment]
Fix released in v0.2.1
